kong/db/strategies/cassandra/init.lua

```diff
@@ -60,6 +60,128 @@ local function extract_pk_values(schema, entity)
 end
 
 
+local function is_partitioned(self)
+  local cql
+
+  if self.connector.major_version == 3 then
+    cql = fmt([[
+      SELECT * FROM system_schema.columns
+      WHERE keyspace_name = '%s'
+      AND table_name = '%s'
+      AND column_name = 'partition'
+      AND kind = 'partition_key'
+      ALLOW FILTERING;
```

Comment:
Instead, it might be better to use:
```
SELECT * FROM system_schema.columns
  WHERE keyspace_name = 'kong_tests'
  AND table_name = 'routes'
  AND column_name = 'partition'
  AND kind = 'partition_key';
```

And when we retrieve the row:
```lua
return column_row.kind == "partition_key"
```

And adapt the C* 2.x table the same way (it can probably avoid this `ALLOW FILTERING` clause as well).

Reply:
@kikito ping. Thoughts on this?

Reply:
Hi, the code you wrote provokes an `ALLOW FILTERING` error in cqlsh:

```
SELECT * FROM system_schema.columns
   ...   WHERE keyspace_name = 'kong_tests'
   ...   AND table_name = 'routes'
   ...   AND column_name = 'partition'
   ...   AND kind = 'partition_key';
InvalidRequest: Error from server: code=2200 [Invalid query] message="Cannot execute this query as it might involve data filtering and thus may have unpredictable performance. If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING"
```

I have worked under the assumption that what you meant was this instead:

```
SELECT * FROM system_schema.columns
  WHERE keyspace_name = 'kong_tests'
  AND table_name = 'routes'
  AND column_name = 'partition';
```

That one works without ALLOW FILTERING.

> And when we retrieve the row:
> ```
> return column_row.kind == "partition_key"
> ```

That works in Cassandra 3 but not in 2, so I made different exit conditions for this as well.

Reply:
Ah yes, that is what I meant (and check the `.kind` column afterwards).